### ISSUE TITLE
Hide empty classes from the QB Model Browser

### ISSUE THREAD

**Issue by heralden** [intermine/redgenes#662]
We should also add a checkbox to show empty classes, so the user can have them if they want (legacy webapp has this).

**Comment by heralden** [comment]
I don't think this is doable with the current web services we have. As an example. FlyMine will hide `Gene > Child Features` *because all childFeatures collections on objects of type Gene are empty*, but with only the mine's JSON data model, you can only tell the total count of the SequenceFeature class (the referenced type) which is 3270774.

From the documentation for [Template Queries](https://intermine.readthedocs.io/en/latest/webapp/template-queries/):
> There are several processes run after the data loading is completed, one of which the objectstore summarisation. This step counts the number of objects of particular classes, identifies any empty references/collections and collects values to be appear in dropdowns in the query builder and templates. The summarisation process also constructs the indexes needed for “type-ahead” autocompletion, this is configured by adding entries to the ObjectStore Summary file.

If we want this feature in Bluegenes, we'll probably need to expose the result of the *objectstore summarisation* as a web service (or possibly add it to the existing model web service JSON).

**Comment by ValWood** [comment]
Hi,  Is there a timescale for this? We are finding it quite difficult to know what data is present and. to make templates even though we know the data well because it isn't super-obvious whether fields are populated or not (quite often we can't tell only from the label).

@kimrutherford 
@manulera 
@ValWood

**Comment by heralden** [comment]
This is quite tricky to implement and we don't have it planned currently. I hope my suggestion in https://github.com/intermine/pombemine/issues/14#issuecomment-1136122752 will help somewhat. I'll talk with Daniela about adding this to the roadmap.

**Comment by ValWood** [comment]
OK, It helps  now I remembered the model browser!
